test/test_torch.py

```diff
@@ -1887,6 +1887,32 @@ def _test_gather(self, cast, test_bounds=True):
         def test_gather(self):
             self._test_gather(self, lambda t: t)
 
+        def test_ravel(self):
+            # Test that flatten returns 1-dim tensor when given a 0-dim tensor
+            zero_dim_tensor = torch.tensor(123)
+            flat0 = zero_dim_tensor.ravel()
+            one_dim_tensor = torch.tensor([123])
+            flat1 = zero_dim_tensor.ravel()
+
+            self.assertEqual(zero_dim_tensor.shape, torch.Size([]))
+            self.assertEqual(flat0.shape, torch.Size([1]))
+            self.assertEqual(one_dim_tensor.shape, torch.Size([1]))
+            self.assertEqual(flat1.shape, torch.Size([1]))
+            self.assertEqual(flat0, one_dim_tensor)
+            self.assertEqual(flat0, flat1)
+            self.assertEqual(flat0.shape, flat1.shape)
+
+            # Test both float tensor and quantized tensor
+            tensors = [torch.randn(5, 5, 5, 5),
+                       torch._empty_affine_quantized([5, 5, 5, 5],
+                                                     scale=2,
+                                                     zero_point=3,
+                                                     dtype=torch.quint8)]
+            for src in tensors:
+                flat = src.ravel()
+                self.assertEqual(flat.shape, torch.Size([625]))
+                self.assertEqual(src.view(-1), flat.view(-1))
+
```

Comment:
There are two cases for torch.ravel(tensor):
1) tensor is a contiguous tensor (e.g., tensor.is_contiguous() returns True). Then tensor.ravel() will return a view of the original tensor
2) tensor is a non-contiguous tensor. Then tensor.ravel() will return a copy of the original tensor.

We haven't tested (2) yet, that would be nice to add here.

Reply:
To check if a Tensor is a view of another tensor, one can check `tensor._base`. For example:

```
x = torch.randn(3, 3)
x_view = x.view(-1)
x_copy = x.clone()
assert x_view._base is x
assert x_copy._base is None # because x_copy isn't a view
```